src/ui/training/utils.ts

```diff
@@ -0,0 +1,67 @@
+import * as xhr from './xhr'
+import { PuzzleSyncData, PuzzleData } from '../../lichess/interfaces/training'
+import router from '../../router'
+import settings from '../../settings'
+import { OfflinePuzzle } from './interfaces'
+
+export function syncPuzzles() {
+  const unsolved = settings.training.unsolvedPuzzles
```

Comment:
You should define a const for `unsolved()` to avoid calling it multiple times (calling it requires access to filesystem).

Reply:
I had to put it in a local variable declared via a `let` rather than a `const` in order to actually save any function calls.